nilearn/datasets/func.py

```diff
@@ -1275,3 +1277,192 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
         X, y, mask_img = _load_mixed_gambles(map(nibabel.load, data.zmaps))
         data.zmaps, data.gain, data.mask_img = X, y, mask_img
     return data
+
+
+def fetch_megatrawls_netmats(data_dir=None, dimensionality=[25, 50, 100, 200, 300],
+                             timeseries=['multiple_spatial_regression', 'eigen_regression'],
+                             matrices=['correlation', 'partial_correlation'],
+                             resume=True, verbose=1):
+    """Downloads and fetches Network Matrices data from MegaTrawls release in HCP.
+
+    This data can be used to predict relationships between imaging data (functional
+    connectivity) and non-imaging behavioural measures such as age, sex, education, etc.
+    The network matrices are estimated from functional connectivity datasets of 461
+    subjects [1]. Full technical details in [2].
+
+    The network matrices denoted as 'netmats' are estimated using full correlation
+    denoted as 'Znet1' or partial correlation with limited L2 regularisation
+    denoted as 'Znet2'. [1]
+
+    .. versionadded:: 0.1.5
+
+    Parameters
+    ----------
+    data_dir: string, default is None, optional
+        Path of the data directory. Used to force data storage in a specified
+        location.
+
+    dimensionality: integer or integers as list, optional
+        Valid dimensions in integers are [25, 50, 100, 200, 300].
+        By default, network matrices data estimated from brain parcellations
+        of all dimensionalities are returned each in a separate dimensional
+        array (n, n).
+        If set to specific dimension, then network matrices related to
+        particular dimension of brain parcellations will be returned. For example,
+        if set as [25, 50] only data corresponding to dimensionality 25 and 50
+        of array (25, 25) and (50, 50) will be fetched.
+
+    timeseries: string or strings as list, optional
+        Valid methods in strings are ['multiple_spatial_regression', 'eigen_regression']
+        By default, network matrices of both types of timeseries signal extraction
+        methods will be returned. Each method has its own matrix array.
+        If set to ['multiple_spatial_regression'], then correlation matrices
+        estimated using spatial regressor based extraction of subject specific
+        timeseries signals will be returned.
+        If set to ['eigen_regression'], then correlation matrices estimated using
+        first principal eigen component based extraction of subject specific
+        timeseries signals will be returned.
+        For full technical details about each method. Refer to [3] [4] [5]
+
+    matrices: string or strings as list, optional
```

Comment:
I tend to use the python types, here, i.e. bool rather than boolean, str rather than string. In this case I would say:

str or list of str, optional